packages/core/src/components/list-slider/list-slider.tsx

```diff
@@ -0,0 +1,260 @@
+  }
+
+  handleNavigationClick(event: Event) {
+    const button = event.currentTarget as HTMLButtonElement
+    const isNext = button?.classList.contains('navigation--next')
+    const isPrev = button?.classList.contains('navigation--prev')
+
+    if (isNext) {
+      this.clickNext.emit()
+    } else if (isPrev) {
+      this.clickPrev.emit()
+    }
+
+    const newIndex = this.currentIndex + (isNext ? 1 : -1)
+
+    const isLastItemAndNext = isNext && newIndex >= this.sliderItems.length
+    const isFirstItemAndPrev = !isNext && newIndex < 0
+
+    if (isLastItemAndNext || isFirstItemAndPrev) return
+
+    this.currentIndex = newIndex
+
+    this.showOrHideNavigationButtons()
+  }
+
+  handleMaxTranslateX() {
+    let totalWidth = 0
+
+    this.viewportWidth = this.sliderWrapper.offsetWidth
+
+    Array.from(this.sliderItems).forEach((item, index) => {
+      totalWidth += item.offsetWidth
+      if (index !== this.sliderItems.length - 1) {
+        totalWidth += this.sliderGapValue
+      }
+    })
+
+    if (this.centralized === true) {
+      this.currentCentralized = totalWidth <= this.viewportWidth
+    }
+
+    this.maxTranslateX = totalWidth - this.viewportWidth
+  }
+
+  showOrHideNavigationButtons() {
+    const hasButtons = this.nextButton && this.prevButton
+
+    if (!hasButtons) return
+
+    const isAtEnd = this.currentTranslateX >= this.maxTranslateX
+    const isAtStart = this.currentTranslateX === 0
+
+    this.nextButton.setAttribute('aria-disabled', String(isAtEnd))
+    this.prevButton.setAttribute('aria-disabled', String(isAtStart))
+  }
+
+  updateSliderPosition() {
+    if (!this.sliderItems || !this.sliderWrapper) return
+
+    const totalWidth = Array.from(this.sliderItems).reduce(
+      (total, item, index) => {
+        let itemTotal = total + item.offsetWidth
+
+        if (index !== this.sliderItems.length - 1) {
+          itemTotal += this.sliderGapValue
+        }
+
+        return itemTotal
+      },
+      0
+    )
+
+    const totalPrevWidth = Array.from(this.sliderItems)
+      .slice(0, this.currentIdx)
+      .reduce((total, item, index) => {
+        let itemTotal = total + item.offsetWidth
+
+        if (index !== this.currentIdx - 1) {
+          itemTotal += this.sliderGapValue
+        }
+
+        return itemTotal
+      }, 0)
+
+    const translateX = Math.min(totalPrevWidth, totalWidth - this.viewportWidth)
+
+    this.sliderWrapper.style.transform = `translateX(-${translateX}px)`
+  }
+
+  handleTouchStart = (event: TouchEvent) => {
+    this.touchStartX = event.touches[0].clientX
+  }
+
+  handleTouchEnd = (event: TouchEvent) => {
+    this.touchEndX = event.changedTouches[0].clientX
+    this.handleSwipe()
+  }
+
+  handleSwipe = () => {
+    const lastIndex = this.sliderItems.length - 1
+    const isSwipeLeft = this.touchEndX < this.touchStartX
+
+    this.currentIndex = isSwipeLeft
+      ? Math.min(this.currentIndex + 1, lastIndex)
+      : Math.max(this.currentIndex - 1, 0)
+  }
```

Comment:
Guess it would be nice to handle all those calculations (mainly in the first load page) in a separate thread. I'm not sure if it is possible, but we can investigate in the future.

Reply:
Do you have any idea how to improve it? We could try in the future

Reply:
I had tried Web Workers but directly in the application: https://dev.to/tassiofront/avoid-overloading-the-main-thread-with-web-workers-557c. I'm not sure if it will be able here as Atomium can be used in the server as well... t will be a nice investigation ahahahah

Reply:
@tassioFront Won't transferring this data between "threads" be worse than calculating directly in the main thread? since it is not very heavy

Reply:
@igorwessel @tassioFront Could u create an issue to improve it later? 🙏

Reply:
that's a nice point, @igorwessel ! haha Certainly it needs a balance! 

> @igorwessel @tassioFront Could u create an issue to improve it later?

Sure! I create it and put here :rocket: